processing/src/main/java/org/apache/druid/segment/join/HashJoinSegmentStorageAdapter.java

```diff
@@ -212,8 +212,11 @@ public Metadata getMetadata()
   @Override
   public boolean hasBuiltInFilters()
   {
-    return clauses.stream()
-                  .anyMatch(clause -> clause.getJoinType() == JoinType.INNER && !clause.getCondition().isAlwaysTrue());
+    // empty clause means that the join has been converted to a filter (captured in baseFilter). So, if the baseFilter
+    // non-null, then the adapter can potentially filter rows.
+    return (clauses.isEmpty() && baseFilter != null) || clauses.stream().anyMatch(
+        clause -> clause.getJoinType() == JoinType.INNER && !clause.getCondition().isAlwaysTrue()
```

Comment:
shouldn't this be 
```suggestion
    // empty clause means that the join has been converted to a filter (captured in baseFilter). So, if the baseFilter
    // non-null, then the adapter can potentially filter rows.
    return (baseFilter != null) || clauses.stream().anyMatch(
        clause -> clause.getJoinType() == JoinType.INNER && !clause.getCondition().isAlwaysTrue()
```

Reply:
if the baseFilter is not null, then rows from underlying storage can be filtered, no matter what is in the clauses.

Reply:
good catch! your suggestion will also accomodate cases where the baseFilter is originating from other places too apart from join to filter conversion.